src/services/embedding/interfaces.ts

```
@@ -0,0 +1,215 @@
+ * Core interfaces for embedding service architecture
+ * Provides separation of concerns with clear boundaries between components
+ */
+
+import { FunctionInfo } from '../../types';
+
+// Model definitions with strict typing
+export const enum EmbeddingModel {
+  ADA_002 = 'text-embedding-ada-002',
+  SMALL_3 = 'text-embedding-3-small',
+  LARGE_3 = 'text-embedding-3-large'
+}
```

Biome (1.9.4)

[error] The enum declaration should not be const

Const enums are not supported by bundlers and are incompatible with the 'isolatedModules' mode. Their use can lead to import inexistent values.
See TypeScript Docs for more details.
Safe fix: Turn the const enum into a regular enum.


(lint/suspicious/noConstEnum)